Cubical/Algebra/Field/Base.agda

```diff
@@ -0,0 +1,183 @@
+{-# OPTIONS --safe #-}
+module Cubical.Algebra.Field.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.SIP
+open import Cubical.Relation.Nullary
+
+open import Cubical.Data.Sigma
+open import Cubical.Data.Unit
+open import Cubical.Data.Empty
+
+open import Cubical.Algebra.Semigroup
+open import Cubical.Algebra.Monoid
+open import Cubical.Algebra.AbGroup
+open import Cubical.Algebra.Ring
+  hiding (_$_)
+open import Cubical.Algebra.CommRing
+
+open import Cubical.Displayed.Base
+open import Cubical.Displayed.Auto
+open import Cubical.Displayed.Record
+open import Cubical.Displayed.Universe
+
+open import Cubical.Reflection.RecordEquiv
+
+open Iso
+
+private
+  variable
+    ℓ ℓ' : Level
+
+record IsField {R : Type ℓ}
+                  (0r 1r : R) (_+_ _·_ : R → R → R) (-_ _⁻¹ : R → R) : Type ℓ where
+
+  constructor isfield
+
+  field
+    isCommRing : IsCommRing 0r 1r _+_ _·_ -_
+    ·⁻¹≡1      : (x : R) → ¬ (x ≡ 0r) → x · (x ⁻¹) ≡ 1r
```

Comment:
I think that if I change this line to this:
```agda
0≡⊎·⁻¹     : (x : R) → (x ≡ 0r) ⊎ (x · (x ⁻¹) ≡ 1r
```

I can't prove this anymore in line 156:
```agda
  isPropInv : isProp ((x : _) → (x ≡ 0r) ⊎ (x · (x ⁻¹) ≡ 1r))
```